src/mmw/js/src/draw/views.js

```diff
@@ -201,6 +188,61 @@ var ResetDrawView = Marionette.ItemView.extend({
     }
 });
 
+function getShapeAndAnalyze(e, model, ofg, grid, tableId) {
+    console.log('clicked');
+    // The shapeId might not be available at the time of the click
+    // because the UTF Grid layer might not be loaded yet, so
+    // we poll for it.
+    var pollInterval = 200,
+        maxPolls = 5,
+        pollCount = 0,
+        deferred = $.Deferred(),
+        shapeId = e.data ? e.data.id : null;
+
+        if (shapeId) {
+            _getShapeAndAnalyze();
+        } else {
+            pollForShapeId();
+        }
+
+    function _getShapeAndAnalyze() {
+        App.restApi.getPolygon({
+            tableId: tableId,
+            shapeId: shapeId
+        }).done(function(shape) {
+            addLayer(shape);
+            ofg.clearLayers();
+            navigateToAnalyze();
+            deferred.resolve();
+        }).fail(function() {
+            console.log('Shape endpoint failed');
+            deferred.reject();
+        }).always(function() {
+            model.enableTools();
+        });
+    }
+
+    function pollForShapeId() {
+        console.log('Shape ID not available yet.');
+        if (pollCount < maxPolls) {
+            var shapeData = grid._objectForEvent(e).data;
+            if (shapeData && shapeData.id) {
+                shapeId = shapeData.id;
+                _getShapeAndAnalyze();
+            } else {
+                window.setTimeout(pollForShapeId, pollInterval);
+                pollCount++;
+            }
+        } else {
+            console.log('Failed to get shape ID within time limit.');
+            model.enableTools();
+            deferred.reject();
```

Comment:
Let's add a popup up here on the click point that apologizes for not being able to select the shape and to try again.